frameworks/legaldocml/web/html-table-support.js

```diff
@@ -0,0 +1,196 @@
+  if (isFrameworkActions(actionsConfiguration)) {
+    var join_action = [
+      {"id": "table.join", "type": "action"}
+    ];
+    var split_action = [
+      {"id": "table.split", "type": "action"}
+    ];
+    var row_actions = [
+      {"id": "insert.table.row.above", "type": "action"},
+      {"id": "insert.table.row.below", "type": "action"},
+      {"id": "delete.table.row", "type": "action"}
+    ];
+    var column_actions = [
+      {"id": "insert.table.column.before", "type": "action"},
+      {"id": "insert.table.column.after", "type": "action"},
+      {"id": "delete.table.column", "type": "action"}
+    ];
+	
+     // Make table-related actions context-aware.
+    [].concat(join_action, split_action, row_actions, column_actions).forEach(function(action) {
+      sync.actions.TableAction.wrapTableAction(editor, action.id);
+    });
+
+    // Wrap the table split action
+    var splitActionId = 'table.split';
+    var originalSplitAction = editor.getActionsManager().getActionById(splitActionId);
+    if (originalSplitAction) {
+      var splitTableAction = new sync.actions.SplitTableCell(
+        originalSplitAction,
+        editor);
+      editor.getActionsManager().registerAction(splitActionId, splitTableAction);
+    }
+
+    var contextualItems = actionsConfiguration.contextualItems;
+    for (var i = 0; i < contextualItems.length; i++) {
+      if (contextualItems[i].name === tr(msgs.TABLE_)) {
+        var items = contextualItems[i].children;
+
+        var row_actions_index = indexOfId(items, row_actions[2].id);
+        goog.bind(items.splice, items, row_actions_index, 1).apply(items, row_actions);
+
+        var column_actions_index = indexOfId(items, column_actions[2].id);
+        goog.bind(items.splice, items, column_actions_index, 1).apply(items, column_actions);
+        break;
+      }
+    }
+  }
+}
+
+/**
+ * @param {Array<{id:string}>} items The array of items.
+ * @param {string} id The ID that we search for.
+ * @return {number} The index of the element with the given ID.
+ */
+function indexOfId(items, id) {
+  for (var i = 0; i < items.length; i++) {
+    if (items[i].id === id) {
+      return i;
+    }
+  }
+  return -1;
+}
+
+/**
+ * @param {object} actionsConfiguration The actions configuration.
+ *
+ * @return {boolean} true if the actions loaded come from the framework.
+ */
+function isFrameworkActions(actionsConfiguration) {
+  var toolbars = actionsConfiguration.toolbars;
+  
+  for( i = 0 ; i < toolbars.length; i++){ 
+	if(toolbars[i].name != "Builtin" && toolbars[i].name != "Review")
+	{
+		return true;
+	}
+	
+  }
+
+}
+
+// Publish the extension.
+sync.ext.Registry.extension = new sync.table.HtmlExtension.Extension();
+
+sync.actions.AkomanNtosoInsertTable = function(a, b, c, d, e, f, g) {
+	 sync.actions.InsertTable.call(this, a, b, c, d, e, f, g);
+	 this.editor = c;
+}
+goog.inherits(sync.actions.AkomanNtosoInsertTable, sync.actions.InsertTable);
```

Comment:
Do we really need to extend the default operation? I think just calling InsertFragmentOpeation would be enough.

Reply:
Only because _sync.actions.InsertTable_ show a dialog that asks for table's title.